tests/testthat/test-input-slider.R

```diff
@@ -0,0 +1,79 @@
+  x <- 10
+  expect_silent(sliderInput('s', 's', x-1, x+1, x))
+  expect_silent(sliderInput('s', 's', x-1, x+1, x-1))
+  expect_silent(sliderInput('s', 's', x-1, x+1, c(x-1, x+1)))
+
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x+2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x-2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x-2, x)))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x, x+2)))
+
+  expect_error(sliderInput('s', 's', x-1,  x+1))
+  expect_error(sliderInput('s', 's', x-1,  x+1,  NULL))
+  expect_error(sliderInput('s', 's', x-1,  NULL, x))
+  expect_error(sliderInput('s', 's', NULL, x+1,  x))
+  expect_error(sliderInput('s', 's', NULL, NULL, x))
+  expect_error(sliderInput('s', 's', x-1,  x+1, NA_real_))
+
+  # Date
+  x <- Sys.Date()
+  expect_silent(sliderInput('s', 's', x-1, x+1, x))
+  expect_silent(sliderInput('s', 's', x-1, x+1, x-1))
+  expect_silent(sliderInput('s', 's', x-1, x+1, c(x-1, x+1)))
+
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x+2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x-2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x-2, x)))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x, x+2)))
+
+  expect_error(sliderInput('s', 's', x-1,  x+1))
+  expect_error(sliderInput('s', 's', x-1,  x+1,  NULL))
+  expect_error(sliderInput('s', 's', x-1,  NULL, x))
+  expect_error(sliderInput('s', 's', NULL, x+1,  x))
+  expect_error(sliderInput('s', 's', NULL, NULL, x))
+  expect_error(sliderInput('s', 's', x-1,  x+1, as.Date(NA)))
+
+  # POSIXct
+  x <- Sys.time()
+  expect_silent(sliderInput('s', 's', x-1, x+1, x))
+  expect_silent(sliderInput('s', 's', x-1, x+1, x-1))
+  expect_silent(sliderInput('s', 's', x-1, x+1, c(x-1, x+1)))
+
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x+2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x-2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x-2, x)))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x, x+2)))
+
+  expect_error(sliderInput('s', 's', x-1,  x+1))
+  expect_error(sliderInput('s', 's', x-1,  x+1,  NULL))
+  expect_error(sliderInput('s', 's', x-1,  NULL, x))
+  expect_error(sliderInput('s', 's', NULL, x+1,  x))
+  expect_error(sliderInput('s', 's', NULL, NULL, x))
+
+  # POSIXLt
+  x <- as.POSIXlt(Sys.time())
+  expect_silent(sliderInput('s', 's', x-1, x+1, x))
+  expect_silent(sliderInput('s', 's', x-1, x+1, x-1))
+  expect_silent(sliderInput('s', 's', x-1, x+1, c(x-1, x+1)))
+
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x+2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  x-2))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x-2, x)))
+  expect_warning(sliderInput('s', 's', x-1,  x+1,  c(x, x+2)))
+
+  expect_error(sliderInput('s', 's', x-1,  x+1))
+  expect_error(sliderInput('s', 's', x-1,  x+1,  NULL))
+  expect_error(sliderInput('s', 's', x-1,  NULL, x))
+  expect_error(sliderInput('s', 's', NULL, x+1,  x))
+  expect_error(sliderInput('s', 's', NULL, NULL, x))
+})
```

Comment:
Let's test for length of inputs.

```suggestion

  # Size
  ## length 0
  expect_error(sliderInput('s', 's', x-1, x+1, numeric(0)))
  expect_error(sliderInput('s', 's', x-1, numeric(0), x))
  expect_error(sliderInput('s', 's', numeric(0), x+1, x))
  ## length 1
  expect_silent(sliderInput('s', 's', x-1, x+1, x))
  ## length 2
  expect_silent(sliderInput('s', 's', x-1, x+1, c(x, x)))
  ## length 3+
  expect_error(sliderInput('s', 's', x-1, x+1, c(x, x, x)))
  expect_error(sliderInput('s', 's', x-1, c(x, x, x), x))
  expect_error(sliderInput('s', 's', c(x, x, x), x+1, x))
})
```

Should we also test for an error if min/max are a vector of values?